slick/src/sphinx/code/Connection.scala

```diff
@@ -103,6 +105,29 @@ object Connection extends App {
       val f: Future[Unit] = db.run(a)
       //#transaction
       Await.result(f, Duration.Inf)
+    };{
+      //#rollback
+      val insertCoffeesAction = coffees ++= Seq (
+        ("Cold_Drip", new SerialBlob(Array[Byte](101))),
+        ("Dutch_Coffee", new SerialBlob(Array[Byte](49)))
+      )
+
+      val countAction = coffees.length.result.map { coffeesCount: Int =>
+        if(coffeesCount < 5000) {
+          DBIO.failed(new Exception())
+        }
+      }
```

Comment:
Ah I see what you mean. It does seem odd in normal code where we want to ensure success, but I actually wanted the `Failure` case to be the result, to highlight that `DBIO.failed` does indeed lead to `Failure`. Does this make sense? If so, then I'll add an `assert` as you suggested, but instead checking for the `Failure` text.